plugin/go.vim

```diff
@@ -6,35 +6,39 @@ let g:go_loaded_install = 1
 
 " these packages are used by vim-go and can be automatically installed if
 " needed by the user with GoInstallBinaries
-let s:packages = [
-      \ "github.com/nsf/gocode",
-      \ "github.com/alecthomas/gometalinter",
-      \ "golang.org/x/tools/cmd/goimports",
-      \ "golang.org/x/tools/cmd/guru",
-      \ "golang.org/x/tools/cmd/gorename",
-      \ "github.com/golang/lint/golint",
-      \ "github.com/rogpeppe/godef",
-      \ "github.com/kisielk/errcheck",
-      \ "github.com/jstemmer/gotags",
-      \ "github.com/klauspost/asmfmt/cmd/asmfmt",
-      \ "github.com/fatih/motion",
-      \ "github.com/fatih/gomodifytags",
-      \ "github.com/zmb3/gogetdoc",
-      \ "github.com/josharian/impl",
-      \ "github.com/dominikh/go-tools/cmd/keyify",
-      \ "github.com/davidrjenni/reftools/cmd/fillstruct",
-      \ ]
+let s:packages = {
+      \ 'asmfmt':        ['github.com/klauspost/asmfmt/cmd/asmfmt'],
+      \ 'errcheck':      ['github.com/kisielk/errcheck'],
+      \ 'fillstruct':    ['github.com/davidrjenni/reftools/cmd/fillstruct'],
+      \ 'gocode':        ['github.com/nsf/gocode', {'windows': '-ldflags -H=windowsgui'}],
+      \ 'godef':         ['github.com/rogpeppe/godef'],
+      \ 'gogetdoc':      ['github.com/zmb3/gogetdoc'],
+      \ 'goimports':     ['golang.org/x/tools/cmd/goimports'],
+      \ 'golint':        ['github.com/golang/lint/golint'],
+      \ 'gometalinter':  ['github.com/alecthomas/gometalinter'],
+      \ 'gomodifytags':  ['github.com/fatih/gomodifytags'],
+      \ 'gorename':      ['golang.org/x/tools/cmd/gorename'],
+      \ 'gotags':        ['github.com/jstemmer/gotags'],
+      \ 'guru':          ['golang.org/x/tools/cmd/guru'],
+      \ 'impl':          ['github.com/josharian/impl'],
+      \ 'keyify':        ['github.com/dominikh/go-tools/cmd/keyify'],
+      \ 'motion':        ['github.com/fatih/motion'],
+\ }
```

Comment:
Do we really want a dictionary here? In every case the key is simply the basename of the first element in the list.

Perhaps this would suffice:
```
let s:packages = {
    \  'github.com/klauspost/asmfmt/cmd/asmfmt': {}`,
...
    \ 'github.com/nsf/gocode': {'windows: '-ldflags -H=windowsgui'},
...
\  }

Food for thought...

Reply:
I started some work on converting this, but it complicates some of the logic by quite a bit (e.g. for filtering the selected packages). I think that keeping it like this is better (a little bit of redundancy in data is better than complicated logic).

I saved the (non-working) patch here: https://gist.github.com/Carpetsmoker/f2003d47f985734c766dff6b68390717